backtester/eventhandlers/statistics/currencystatistics/currencystatistics.go

```diff
@@ -0,0 +1,295 @@
+package currencystatistics
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/thrasher-corp/gocryptotrader/backtester/common"
+	gctcommon "github.com/thrasher-corp/gocryptotrader/common"
+	"github.com/thrasher-corp/gocryptotrader/common/math"
+	"github.com/thrasher-corp/gocryptotrader/currency"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/asset"
+	gctkline "github.com/thrasher-corp/gocryptotrader/exchanges/kline"
+	gctorder "github.com/thrasher-corp/gocryptotrader/exchanges/order"
+	"github.com/thrasher-corp/gocryptotrader/log"
+)
+
+// CalculateResults calculates all statistics for the exchange, asset, currency pair
+func (c *CurrencyStatistic) CalculateResults() error {
+	var errs gctcommon.Errors
+	first := c.Events[0]
+	firstPrice := first.DataEvent.ClosePrice()
+	last := c.Events[len(c.Events)-1]
+	lastPrice := last.DataEvent.ClosePrice()
+	for i := range last.Transactions.Orders {
+		if last.Transactions.Orders[i].Side == gctorder.Buy {
+			c.BuyOrders++
+		} else if last.Transactions.Orders[i].Side == gctorder.Sell {
+			c.SellOrders++
+		}
+	}
+	for i := range c.Events {
+		price := c.Events[i].DataEvent.ClosePrice()
+		if c.LowestClosePrice == 0 || price < c.LowestClosePrice {
+			c.LowestClosePrice = price
+		}
+		if price > c.HighestClosePrice {
+			c.HighestClosePrice = price
+		}
+	}
+	c.MarketMovement = ((lastPrice - firstPrice) / firstPrice) * 100
+	c.StrategyMovement = ((last.Holdings.TotalValue - last.Holdings.InitialFunds) / last.Holdings.InitialFunds) * 100
+	c.RiskFreeRate = last.Holdings.RiskFreeRate * 100
+	returnPerCandle := make([]float64, len(c.Events))
+
+	var allDataEvents []common.DataEventHandler
+	for i := range c.Events {
+		returnPerCandle[i] = c.Events[i].Holdings.ChangeInTotalValuePercent
```

Comment:
I'm not so sure on that suggestion. Im also unsure on what is wrong with `ChangeInTotalValuePercent`, is `TotalValuePercentageChange` clearer?. Perhaps is it that `Events` is confusing and so seeing a change in total value percent looks weird? I could call it IntervalEvents so that its clear that the total change in value percentage is related to an interval?

Reply:
I'm not sure what exactly I was confused on anymore. Maybe on the next review I'll see if I come across the same thing again.  Seems all good for now. No need to change it